src/Auth0.ManagementApi.Client/ApiConnection.cs

```diff
@@ -125,6 +128,13 @@ public ApiConnection(string token, string baseUrl)
                 }
             }
 
+            // Add the diagnostics header, unless user explicitly opted out of it
+            if (!object.ReferenceEquals(diagnostics, DiagnosticsHeader.Suppress))
+                request.AddHeader("Auth0-Client", diagnostics.ToString());
+
+            // Add the user agent
+            request.AddHeader("User-Agent", ".NET/PCL");

```

Comment:
PortableRest automatically adds a UserAgent header. If you call SetUserAgent<ClientBase>() in your constructor, it will automatically add a User Agent header with both the current assembly version, and the version of PortableRest used. See https://github.com/advancedrei/PortableRest/blob/master/src/PortableRest/RestClient.cs#L133-L171

Also, there is a way to use Interfaces and a technique with PCLs called "bait and switch" to leverage NuGet to be able to swap out portable libraries with platform specific libraries. That technique could be used to get more specific diagnostic information. However, any diagnostics information sent down should be opt-in. I don't think just because I'm using your library means you have the right to scrape my system for information without my permission. Something to keep in mind.